SWLOR.Game.Server/Feature/GuiDefinition/ViewModel/HoloNetViewModel.cs

```diff
@@ -0,0 +1,92 @@
+﻿using System;
+using System.Threading.Tasks;
+using Discord;
+using Discord.Webhook;
+using SWLOR.Game.Server.Service;
+using SWLOR.Game.Server.Service.GuiService;
+
+namespace SWLOR.Game.Server.Feature.GuiDefinition.ViewModel
+{
+    public class HoloNetViewModel : GuiViewModelBase<HoloNetViewModel, GuiPayloadBase>
+    {
+        public string HoloNetText
+        {
+            get => Get<string>();
+            set => Set(value);
+        }
+
+        public const int MaxHoloNetTextLength = 800;
+        private const int Price = 2500;
+
+        protected override void Initialize(GuiPayloadBase initialPayload)
+        {
+            HoloNetText = string.Empty;
+            WatchOnClient(model => model.HoloNetText);
+        }
+
+        public Action OnClickSubmit() => () =>
+        {
+            if (string.IsNullOrWhiteSpace(HoloNetText))
+            {
+                return;
+            }
+
+            var message = HoloNetText;
+
+            if (message.Length > 600)
+            {
+                SendMessageToPC(Player, "Your HoloNet broadcast was too long. Please shorten it to no longer than 600 characters and resubmit the broadcast. For reference, your message was: \"" + message + "\"");
```

Comment:
The message says 600 characters is the limit but your constant on line 18 is 800. Would recommend replacing the 600 here with MaxHoloNetTextLength

In C# this can be done like so:
`
SendMessageToPC(Player, $"Your HoloNet broadcast was too long. Please shorten it to no longer than {MaxHoloNetTextLength} characters and resubmit the broadcast. For reference, your message was: \"" + message + "\"");`

The $ before the opening quote indicates that you can embed string values inside the string. Then you can put { } around the code value you want to embed.

Reply:
Added